### PR TITLE
Request page

### PR DESCRIPTION
**Pull request by Rawan2211**

### CODE REVIEW

**Review by yaay — APPROVED** [review]

File: src/app/pages/application-status/application-status.component.ts [hunk 1]
```diff
@@ -1,17 +1,58 @@
...
   styleUrls: ['./application-status.component.css']
 })
 export class ApplicationStatusComponent {
+  statusValue:string="";
+  requestValue:string = "";
+  showBtn:boolean=true;
 
-  constructor(private router: Router) {}
+  constructor(private router: Router,private reqStatusRepository:RequestStatusRepository,
+    private ReqRepository:RequestRepository) {}
```
Comment (Tasneem-Gad) [inline src/app/pages/application-status/application-status.component.ts:17]: we use camel case

File: src/app/pages/application-status/application-status.component.ts [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
 
   startApplication() {
```
Comment (Tasneem-Gad) [inline src/app/pages/application-status/application-status.component.ts:19]: naming must be meaning full

File: src/app/pages/application-status/application-status.component.ts [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
   }
+  ngOnInit(){
+    this.startbtn();
+  }
+  startbtn(){
+    this.reqStatusRepository.get().subscribe({
+      error: () =>
+      {
+        this.requestValue = "بدأ الطلب",
+        this.statusValue="تقديم الطلب"
```
Comment (Tasneem-Gad) [inline src/app/pages/application-status/application-status.component.ts:41]: why made status static

File: src/app/pages/application-status/application-status.component.ts [hunk 4]
```diff
@@ -1,17 +1,58 @@
...
+      },
+      next: (res) => {
+      if (res.requestStatus.code == "DRAFT")
+      {
+        this.requestValue = "غير مكتمل"
```
Comment (Tasneem-Gad) [inline src/app/pages/application-status/application-status.component.ts:46]: same

File: src/app/pages/application-status/application-status.component.ts [hunk 5]
```diff
@@ -1,17 +1,58 @@
...
+        this.statusValue="استكمال الطلب";
+      }
+        else if(res.requestStatus.code == "NEW_REQUEST")
+        {
+          this.showBtn=false;
+          this.requestValue = "تم تقديم الطلب"
```
Comment (Tasneem-Gad) [inline src/app/pages/application-status/application-status.component.ts:52]: new request mean apply new request not complete

**Review by Tasneem-Gad — CHANGES_REQUESTED** [review]

**Pull request merged**